src/index.ts

```
@@ -32,29 +20,83 @@ class SimpleOpenFile {
     const end = data.byteLength;
     const slice = data.slice(start, end);
     this.file_pos = end;
-    const string = new TextDecoder().decode(slice);
-    return string;
+    return new TextDecoder().decode(slice);
   }
 }
 
-// @ts-expect-error
-let stderr = new SimpleOpenFile(wasi.fds[2].file);
+const defaultBaseUrl = "https://compbio.fhs.um.edu.mo/diffeq";
 
-// @ts-expect-error
-let stdout = new SimpleOpenFile(wasi.fds[1].file);
+function createWasi() {
+  let args: string[] = [];
+  let env: string[] = [];
+  let fds = [
+    new OpenFile(new File([])), // stdin
+    new OpenFile(new File([])), // stdout
+    new OpenFile(new File([])), // stderr
+  ];
+  return new WASI(args, env, fds);
+}
 
-function getWasmMemory() {
-  if (inst === undefined) {
-    throw new Error("WASM module not loaded");
+class ModelRegistry {
+  private static models: Map<string, any> = new Map();
+  private static defaultModel: any = null;
+
+  static register(id: string | undefined, model: any) {
+    if (id) {
+      this.models.set(id, model);
+    } else {
+      this.defaultModel = model;
+    }
+  }
+
+  static get(id?: string) {
+    if (id) {
+      return this.models.get(id);
+    }
```

if an `id` is supplied but not in the registry, you might want to throw an error here. I can imagine that it would be easy to just have a typo in your id name and you would want to know about that rather than just get the default model